transport/api/src/p2p.rs

```diff
@@ -160,14 +174,37 @@ pub async fn p2p_loop(
             Ok(ma) => {
                 if let Err(e) = swarm.listen_on(ma.clone()) {
                     error!("Failed to listen_on using the multiaddress '{}': {}", multiaddress, e);
+
+                    match replace_transport_with_unspecified(&ma) {
+                        Ok(ma) => {
+                            if let Err(e) = swarm.listen_on(ma.clone()) {
+                                error!(
+                                    "Failed to listen_on also using the unspecified multiaddress '{}': {}",
+                                    ma, e
+                                );
+                            } else {
+                                info!("Successfully started listening on {ma} (from {multiaddress})");
+                                swarm.add_external_address(multiaddress.clone());
+                            }
+                        }
+                        Err(e) => {
+                            error!("Failed to transform the multiaddress '{ma}' to unspecified: {e}")
+                        }
+                    }
                 } else {
-                    info!("Successfully started listening on {ma} (from {multiaddress})")
+                    info!("Successfully started listening on {ma} (from {multiaddress})");
+                    swarm.add_external_address(multiaddress.clone());
```

Comment:
please add a check if it is a public address, could be a private one

Reply:
Can we do it in a future cleanup?
1. the add external address is not used anywhere and I put it there just to be consistent with your example.
2. mulitaddress checks for ip's are already performed one layer above this.